### ISSUE TITLE
Make build_settings available without initializing ApplicationContext

### ISSUE THREAD

**Issue by hansonap** [mherrmann/fbs#224]
It is sometimes necessary to get information in build_settigns before initializing a class that inherits from ApplicationContext, but because ApplicationContext.build_settings is a property, this is not possible.

This could be remedied by:
1. Converting property to static method
2. Converting property to function
3. Create function which the property calls, but can also be called by the user.

For now, it seems the below is an acceptable workaround (Please comment on any unforeseen drawbacks):
```python
def init_logging():
    """
    We init logging here, outside of main.AppContext, so that local DEBUG logging and global ERROR logging can 
    occur before initializing AppContext.
    """
    build_settings = ApplicationContext().build_settings  # this works because _state.APPLICATION_CONTEXT is still NONE
    logging.setLoggerClass(app_logging.Logger)
    ...
    formatter = app_logging.Formatter(build_settings['version'], uuid)
    ...
```

If you agree, I'd be happy to create pull request with solution. My preference is 3. so it doesn't impact existing implementations and the convenience is maintained.

**Comment by mherrmann** [comment]
Sorry, I don't understand why specifically you give `init_logging()` as an example. Is it because that's what you'd overwrite in the PR? Or does it just serve as an example usage of `ApplicationContext().build_settings`? I don't understand what you mean, am very busy and don't have time to attempt to figure out the details. 

Could you please provide a self-contained example of what you would like to do, how it doesn't work, and how you would fix it in fbs?

**Comment by whattheserver** [comment]
I'm pretty sure he just wants a way to use the build settings dictionary without having to call an ApplicationContext instance. he just simulated a basic thing and the logging was just an unrelated example of why he would have called it that way.

In my case i actually like in the other example i posted earlier would really benefit from it being accessible directly via an import.

I can get access to all these  if i do

**Comment by mherrmann** [comment]
I implemented this now in fbs Pro 0.9.7. It's called `PUBLIC_SETTINGS` and is documented in the [Manual](https://build-system.fman.io/manual/#public_settings).
